### DISCUSSION TITLE
Multisource Stream Option

### DISCUSSION THREAD

**Discussion by danrossi** [OvenMediaLabs/OvenMediaEngine#1655] [Help]
I think I saw in the dev server option something about multisource muxing. Is it possible to publish to a main source and webrtc gets them from track events. Or programatically add a  stream to a source. Millicast and Ant can do this. It might mean my rooms signal server feature notifying from the webhook might work different. Subscribing to the main source will pickup the tracks directly requiring less sockets open.

**Comment by getroot** [comment]
I didn't fully understand your question. Could you elaborate a bit more?

For Multiplex Channel, please refer to the manual below.

https://airensoft.gitbook.io/ovenmediaengine/live-source/multiplex-channel

You can also create a stream by pulling another stream with Scheduled Channel.

https://airensoft.gitbook.io/ovenmediaengine/live-source/scheduled-channel

**Reply by danrossi** [reply]
I was wondering if that feature works like this. You publish webrtc to a main source. And when subscribing to that main source via webrtc it gets all the tracks by the track event. And the api can add/remove source streams from a second main source. Ant media has a similar thing. It might prevent me needing to do webhooks and the subscriber gets the tracks automatically. A manual config isn't great, might have to be all done from the api to add / remove source streams. Each publisher will have to do an api method to add itself to this channel after webrtc publishing has started ?  They would be all passthrough audio and video. Might still need a webhook for RTMP to make api calls also to join this channel also.

https://docs.dolby.io/streaming-apis/docs/source-and-layer-selection

**Reply by getroot** [reply]
I think OvenMediaEngine's Mux Channel has similar functionality to the link you provided. Our Mux Channel also provides an API, and you can use Mux Channel to take tracks from multiple live streams (whether they are generated from RTMP or WebRTC or whatever) and create a new stream.

And what I find hard to understand is, in your sentence, I don't understand what the concept of RTMP joining this channel is. I think it's a concept used by other services.

"Might still need a webhook for RTMP to make api calls also to join this channel."

**Reply by getroot** [reply]
To help you understand, here are some concepts:

Users can create streams in OME using encoders (like OBS) or mux channels or scheduled channels. The module that does this is called a Provider. OME has RTPM Providers, SRT Providers, Mux Channel Providers, etc.

And the module that streams it so that viewers can play it is called a Publisher. OME has HLS, LLHLS, and WebRTC Publishers.

Any stream created through a Provider can be streamed through a Publisher, regardless of which Provider it was created from.

A Mux Channel is a Provider. It has no subscription or such concept.
